nmdc_automation/re_iding/base.py

```diff
@@ -728,12 +762,79 @@ def _update_omics_processing_alternative_identifiers(omics_processing: nmdc.Omic
         logging.warning(f"Unknown legacy ID format: {legacy_omics_processing_id}")
     return omics_processing
 
+def update_omics_output_data_object(
+        data_object: nmdc.DataObject, nmdc_omics_processing_id: str,
+        api_client: NmdcRuntimeApi, identifiers_map: dict=None) -> nmdc.DataObject:
+    """
+    Update the data object record with the new ID, and add the legacy ID to the alternate identifiers
+    """
+    updated_data_object = deepcopy(data_object)
+    # Check if we need to update the data object ID and set the alternative identifiers to the legacy ID
+    if not updated_data_object.id.startswith("nmdc:dobj-"):
+        updated_data_object.alternative_identifiers = [data_object.id]
```

Comment:
data object omics has_output alternative identifiers should only be set for nom,metabolomics. Proteomics omics has_output also have alternative identifiers but those are being added as entirely new records.

Reply:
https://github.com/microbiomedata/nmdc_automation/pull/116/commits/8d48f5cf35127ba6f231046d9f0ebd171b2573cc